mlflow/pipelines/steps/split.py

```diff
@@ -213,7 +213,19 @@ def _run(self, output_directory):
                 importlib.import_module(post_split_module_name), post_split_fn_name
             )
             _logger.debug(f"Running {post_split_fn_name} on train, validation and test datasets.")
-            (train_df, validation_df, test_df) = post_split(train_df, validation_df, test_df)
+            (post_filter_train_df, post_filter_validation_df, post_filter_test_df) = post_split(
+                train_df, validation_df, test_df
+            )
+            self._validate_user_code_output(train_df, post_filter_train_df)
+            self._validate_user_code_output(validation_df, post_filter_validation_df)
+            self._validate_user_code_output(test_df, post_filter_test_df)
```

Comment:
Can we move these into a single function, something like `_validate_post_split_output`?